pkg/controller/core/workload_controller.go

```diff
@@ -264,8 +272,9 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		reportWorkloadEvictedOnce := workload.WorkloadEvictionStateInc(&wl, kueue.WorkloadDeactivated, "")
 		updated = workload.ResetChecksOnEviction(&wl, r.clock.Now()) || updated
 		if updated {
-			if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock); err != nil {
-				return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
+			err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
+			if client.IgnoreNotFound(err) != nil {
+				return ctrl.Result{}, err
```

Comment:
```suggestion
				return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
```

This is not related changes.

Reply:
Done.